src/sick_tim310.cpp

```diff
@@ -50,19 +50,26 @@ int main(int argc, char **argv)
 
   sick_tim::SickTim310Parser* parser = new sick_tim::SickTim310Parser();
 
-  sick_tim::SickTimCommon* s;
-  if (subscribe_datagram)
-    s = new sick_tim::SickTimCommonMockup(parser);
-  else
-    s = new sick_tim::SickTimCommonUsb(parser);
+  sick_tim::SickTimCommon* s = NULL;
 
-  int result = s->init();
-  while (ros::ok() && (result == EXIT_SUCCESS))
+  int result = EXIT_FAILURE;
+  while (ros::ok())
   {
-    ros::spinOnce();
-    result = s->loopOnce();
+    // Atempt to connect/reconnect
+    delete s;
+    if (subscribe_datagram)
+      s = new sick_tim::SickTimCommonMockup(parser);
+    else
+      s = new sick_tim::SickTimCommonUsb(parser);
+    result = s->init();
+
+    while(ros::ok() && (result == EXIT_SUCCESS)){
+      ros::spinOnce();
+      result = s->loopOnce();
+    }
   }
 
   delete s;
+  delete parser;
```

Comment:
We don't need to delete the parser here, since it's already done in [sick_tim_common.cpp#L89](https://github.com/uos/sick_tim/blob/8288311ac87103759dbffba252197d1efefd523b/src/sick_tim_common.cpp#L89). But perhaps it's more obvious to move the deletion here? Any thoughts? (this also applies to all the other nodes)